assignments/array-methods.js

```diff
@@ -58,28 +58,71 @@ const runners = [
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs. Combine both the first and last names and populate a new array called `fullNames`. This array will contain just strings.
 let fullNames = [];
+  
+runners.forEach(function(items){
+  return fullNames.push(`${items.first_name} ${items.last_name}`)
+});
 console.log(fullNames);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runners' first names in uppercase because the director BECAME DRUNK WITH POWER. Populate an array called `firstNamesAllCaps`. This array will contain just strings.
 let firstNamesAllCaps = [];
+
+firstNamesAllCaps = runners.map(function(item){
+  return item.first_name.toUpperCase();
+ });
+  
 console.log(firstNamesAllCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue. We need a filtered version of the runners array, containing only those runners with large sized shirts so they can choose a different size. This will be an array of objects.
 let runnersLargeSizeShirt = [];
+
+runnersLargeSizeShirt = runners.filter(function(item){
+ //true or false 
+   return item.shirt_size === "L";
+ });
+ 
 console.log(runnersLargeSizeShirt);
 
+
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations and save the total into a ticketPriceTotal variable.
 let ticketPriceTotal = 0;
+
+ticketPriceTotal = runners.reduce(function(accumulator, item){
+  return accumulator + item.donation; 
+}, 0);
+
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+// The run has been moved to a different date due to rain. The event director needs all the runners emails in order to email them about moving the date.
+let emailAddress = [];
+  
```

Comment:
Here, you could use `.map()` instead of `forEach()`.

If you used `.map()`, how would you type the code so that only the emails exist in the new array?